### PR TITLE
Handling of PDF sources

### PR DESCRIPTION
**Pull request by curzio-della-santa**
Due to automatic processing of blitline from src to input.png with PDF files, added specific prams for this conversion to make it compatible with our needs

### CODE REVIEW

File: examples/job-submit.php [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 $job = $jobBuilder->createJob()
     ->setSource(
-        new UrlSource($imageUrl)
+        AwsS3Source::fromUrl($imageUrl)
```
Comment (ivan-wolf) [inline examples/job-submit.php]: What does this change have to do with the PDF preview?

File: src/Detail/Blitline/Client/BlitlineClient.php [hunk 2]
```diff
@@ -182,6 +182,16 @@ public function submitJob($job = array())
                 break;
         }
 
+        // If source is a PDF add specific conversion parameters to process only first page and avoid color loss
+        if ('pdf' === strtolower(pathinfo(parse_url($source->getUrl(), PHP_URL_PATH), PATHINFO_EXTENSION))) {
```
Comment (ivan-wolf) [inline src/Detail/Blitline/Client/BlitlineClient.php]: Can't this fail?

File: src/Detail/Blitline/Job/Definition/SourceDefinition.php [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * @var array
+     */
+    protected $options = array(
+        self::OPTION_NAME   => 's3',
+    );
+
+    /**
+ * @inheritdoc
```
Comment (curzio-della-santa) [inline src/Detail/Blitline/Job/Definition/SourceDefinition.php]: Fix indentation

**Comment by curzio-della-santa** [comment]
Example runs:

- Source as string (Url):
2017-06-14 08_38_40-blitline examples web01 detailnet me_job-submit php_image_url https___s3-eu-west

- Source as SourceDefinition (S3 bucket/key)
2017-06-14 08_38_29-blitline examples web01 detailnet me_job-submit-s3 php_image_key test-files_colo

Here the results correctly generated:

2017-06-14 08_37_33-s3 management console

The pdf has been correctly converted (only first page and colors are correct):
color-management pdf-200_blitline 1

File: src/Detail/Blitline/Job/Definition/JobDefinition.php [hunk 4]
```diff
@@ -2,62 +2,95 @@
...
     const OPTION_FUNCTIONS    = 'functions';
+    const OPTION_SOURCE_TYPE  = 'src_type';
+    const OPTION_SOURCE_DATA  = 'src_data';
 
     /**
      * @var array
      */
     protected $options = array(
-        self::OPTION_VERSION   => '1.21',
+        self::OPTION_VERSION   => '1.22',
```
Comment (ivan-wolf) [inline src/Detail/Blitline/Job/Definition/JobDefinition.php:18]: The documentation says 1.21 is the latest (see http://docs.blitline.com/articles/jobs.html). Is 1.22 correct?

Comment (curzio-della-santa) [reply src/Detail/Blitline/Job/Definition/JobDefinition.php:18]: Is the feedback I've received from Blitline support for the PDF processing "https://www.blitline.com/v3/home/gist?gist_id=7bcc632672e38bcf5e2a4d01899ca9cd"

I've not tried if it works **only** with version 1.22

Comment (ivan-wolf) [reply src/Detail/Blitline/Job/Definition/JobDefinition.php:18]: OK, must be correct then.

**Pull request merged**